llvm/lib/Target/AArch64/AArch64Subtarget.cpp

```diff
@@ -584,6 +584,10 @@ AArch64Subtarget::getAuthenticatedLRCheckMethod() const {
   return AArch64PAuth::AuthCheckMethod::None;
 }
 
+bool AArch64Subtarget::enableSubRegLiveness() const {
+  return hasSME() && isStreaming();
```

Comment:
nit: a function must have SME in order to be streaming, so `hasSME()` is implied and can therefore be removed.